pkg/client/api.go

```diff
@@ -81,20 +81,39 @@ func (c *httpAPIClient) Do(ctx context.Context, verb, endpoint string, query url
 	}()
 
 	if err != nil {
-		return APIResponse{}, err
+		return APIResponse{}, &Error{
+		  	Type: ErrBadResponse,
+		  	ErrorMsg:  err.Error(),
+			Query: query.Encode(),
+			StatusCode: 0,
+		}
 	}
 
 	if klog.V(6).Enabled() {
 		klog.Infof("%s %s %s", verb, u.String(), resp.Status)
 	}
 
+	// Read and decode the response body for logging
+	var respBodyStr string
+	if resp != nil {
+		respBodyBytes, readErr := io.ReadAll(resp.Body)
+		if readErr != nil {
+		  respBodyStr = fmt.Sprintf("unable to read response body: %v", readErr)
+		} else {
+		  respBodyStr = string(respBodyBytes)
+		  // Recreate the response body for further use
+		  resp.Body = io.NopCloser(bytes.NewReader(respBodyBytes))
+		}
+	}
```

Comment:
Should this go in lines 120-129 instead? Seems redundant + we can remove the conditional that verbosity level 8 is enabled

> 	var body io.Reader = resp.Body
> 	if klog.V(8).Enabled() {
> 		data, err := io.ReadAll(body)
> 		if err != nil {
> 			return APIResponse{}, fmt.Errorf("unable to log response body: %v", err)
> 		}
> 		klog.Infof("Response Body: %s", string(data))
> 		body = bytes.NewReader(data)
> 	}